download-mods.ts

```diff
@@ -6,14 +6,13 @@ import { parseArguments, readUri } from "./libs/utils";
 async function main(argc: number, argv: string[])
 {
 	const [fixedArguments] = parseArguments(argc, argv);
-	let argumentsValid = fixedArguments.length == 3;
-	if (!argumentsValid)
+	if (fixedArguments.length != 3)
 	{
 		console.error("Usage: ts-node download-mods.ts <directory> <minecraft-version> <mods.md>");
 		process.exit(1);
 	}
 
-	const serverDirectory  = fixedArguments[0];
+	const modsDirectory  = fixedArguments[0];
```

Comment:
I don't want to accept the rest of the changes because:
* Mod download now involves another billion http requests
* Incremental download now involves a billion http requests even if the hackmd has not changed.
* Downloading the same markdown twice will no longer always produce the same downloads.

Reply:
About these:

- Mod download involves a billion HTTP requests because of recursive dependency resolution, although that can be not done by removing the dependency resolution lines from `downloads.ts`
- Incremental download involves a billion HTTP requests because dependencies *can* change even if the hackmd has not
- Downloading the same markdown will produce the same downloads if the mods' dependencies haven't changed at curseforge

Reply:
I believe the problem of excessive HTTP requests can be solved if the file upload time at curseforge was considered and dependencies were stored in the manifest along with the download time of the mod.

That way the dependencies would only be retrieved and updated if the upload time of the file at curseforge was after the time stored on the manifest.